### PR TITLE
Use `zod` for `FormData` validation

### PR DESCRIPTION
**Pull request by hangler**
For now, this is an example using the "create Bank citation" form. Still somewhat verbose, but better than what we were using before. Will throw a nice error that's catchable.

### CODE REVIEW

**Comment by hangler** [comment]
Sorry @Shawk031, it's a big and ugly PR because I was doing quite a bit of refactoring. I'm afraid there are more like this to come. But, I'll try to break them up more!

**Comment by Shawk031** [comment]
LOL yeah no worries-- ill look at this right now

File: app/routes/bank/create.tsx [hunk 1]
```diff
@@ -1,110 +1,122 @@
...
+import { json, type ActionArgs, redirect } from "@remix-run/server-runtime"
+import { PageHeader } from "~/components/elements/PageHeader"
+import { prisma } from "~/db.server"
+import { z } from "zod"
+import BankEditCitationFields from "~/components/bank/BankEditCitationFields"
+import BankSourcePanel from "~/components/bank/BankSourcePanels/BankSourcePanel"
+import Button from "~/components/elements/Button"
+import invariant from "tiny-invariant"
+
+const BankCreateFormDataSchema = z.object({
```
Comment (Shawk031) [inline app/routes/bank/create.tsx:19]: this library seems so useful

**Review by Shawk031 — APPROVED** [review]
looks good!

**Pull request merged**